src/gravel/controllers/kv.py

```diff
@@ -46,6 +46,7 @@ async def release(self) -> None:
 class KV:
 
     _client: Optional[aetcd3.Etcd3Client]
+    _is_open: bool
```

Comment:
This already existed but was underutilised. It could be redundant if checking `_client` instead. But for this change this was mostly just a tidy up of what was there. It did make the Fake class easier to implement but the fake class could just the same still create one.

I'd rather leave this as a tidy up for now of the existing implementation. If we want to change the state machine we can do it as a followup.